Cart-API/src/routes/index.ts

```diff
@@ -0,0 +1,77 @@
+import express from "express";
+const routes = express.Router();
+// adding this from chatgpt
+const cors = require('cors');
+
+const cartItems = (id: number, product: string, price: number, quantity: number) => ({
+    id,
+    product,
+    price,
+    quantity
+});
+
+const sneakers = [
+    cartItems(1, "Nike Air Jordan I", 124.99, 23),
+    cartItems(2, "Nike Kobe V Protro", 249.99, 8),
+    cartItems(3, "Nike Air More Uptempo", 149.99, 33),
+    cartItems(4, "Puma OG Palace Guard", 74.99, 11),
+    cartItems(5, "Adidas AdiZero Rose 1", 89.99, 1)
+];
+// added from chatgpt
+routes.use(cors());
+
+const filterItemsByPrefix = (sneakers: any[], prefix: string) =>
+    prefix ? sneakers.filter(item => item.product.startsWith(prefix)) : sneakers;
+
+// routes.get("/", (req, res) => {
+//     res.json({hello: "world"})
+// });
+
+// routes.get("/sneakers", (req, res) => {
+//     res.json(sneakers)
+// });
+// routes.get("/sneakers/:id", (req, res) => {
+//     res.json(sneakers[+req.params.id])
+// });
+
+routes.get("/cart-items", (req, res) => {
+    const prefix: string = req.query.prefix as string;
+    const filteredItems = filterItemsByPrefix(sneakers, prefix);
+    res.status(200).json(filteredItems);
+});
+// routes.get("/cart-items/:id", (req, res) => {
+//     res.json(sneakers[+req.params.id])
+// });
+routes.get("/cart-items/:id", (req, res) => {
+    const itemId = +req.params.id;
+    const item = sneakers.find(item => item.id === itemId)
+    if (item) {
+        res.status(200).json(item);
+    } else {
+        res.status(404).json({message: "ID Not Found"});
+    }
+});
+routes.post("/cart-items", (req, res) => {
+    sneakers.push(cartItems(req.body.id, req.body.product, req.body.price, req.body.quantity))
```

Comment:
The issue with this is that you seem to be getting the ID from the body of the request. You could easily end up with duplicate ids. Usually we want to create our body here on the back end, where we can make sure it's unique from all other ids. 

An easy way to do this is to install [the uuid package. ](https://www.npmjs.com/package/uuid) When you import this package at the top of a file, you can call it whenever you need a new, unique ID.